modules/portal/app/controllers/FrontendController.scala

```
@@ -16,50 +16,78 @@
 
 package controllers
 
-import actions.FrontendAction
+import actions.{LdapFrontendAction, OidcFrontendAction}
 import javax.inject.{Inject, Singleton}
 import models.{CustomLinks, Meta}
+import org.pac4j.core.profile.CommonProfile
+import org.pac4j.play.scala.{Pac4jScalaTemplateHelper, Security, SecurityComponents}
 import org.slf4j.LoggerFactory
 import play.api.Logger
 import play.api.mvc._
 import play.api.Configuration
+import vinyldns.core.crypto.CryptoAlgebra
+import vinyldns.core.domain.membership.User
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
-
 /*
  * Controller for specific pages - sends requests along to views
  */
 @Singleton
 class FrontendController @Inject()(
-    components: ControllerComponents,
+    val controllerComponents: SecurityComponents,
     configuration: Configuration,
-    userAccountAccessor: UserAccountAccessor)
-    extends AbstractController(components) {
-
-  private val userAction = Action.andThen(new FrontendAction(userAccountAccessor.get))
+    userAccountAccessor: UserAccountAccessor,
+    crypto: CryptoAlgebra)(
+    implicit val pac4jScalaTemplateHelper: Pac4jScalaTemplateHelper[CommonProfile])
+    extends Security[CommonProfile] {
+
+  val oidcEnabled: Boolean = configuration.getOptional[Boolean]("oidc.enabled").getOrElse(false)
+  lazy val oidcUsernameField: String =
+    configuration.getOptional[String]("oidc.jwt-username-field").getOrElse("username")
+
+  private val vinyldnsServiceBackend =
+    configuration
+      .getOptional[String]("portal.vinyldns.backend.url")
+      .getOrElse("http://localhost:9000")
```

we probably shouldn't do optional here.  I don't know how destructive things will be if someone accidentally omitted the `portal.vinyldns.backend.url` parameter in a real production environment.  Better to assume it is there, make sure it is in the local.conf (which it should be).  If it isn't there let the app blow up

---

this is how weve been doing it historically. can move to non-optional though